unstructured/partition/auto.py

```diff
@@ -274,3 +284,30 @@ def file_and_type_from_url(
 
     filetype = detect_filetype(file=file, content_type=content_type, encoding=encoding)
     return file, filetype
+
+
+def decide_table_extraction(
+    filetype: Optional[FileType],
+    skip_infer_table_types: List[str],
+    pdf_infer_table_structure: bool,
+) -> bool:
+    FILETYPE_MAPPING = {
```

Comment:
Instead of this mapping you could do `filetype.name.lower()`, then we won't have to remember to add to this each time.